stan/math/prim/scal/fun/lower_reg_inc_gamma.hpp

```diff
@@ -0,0 +1,194 @@
+#ifndef STAN_MATH_PRIM_SCAL_FUN_LOWER_REG_INC_GAMMA_HPP
+#define STAN_MATH_PRIM_SCAL_FUN_LOWER_REG_INC_GAMMA_HPP
+
+#include <boost/math/tools/promotion.hpp>
+#include <stan/math/prim/scal/fun/lgamma.hpp>
+#include <stan/math/prim/scal/fun/gamma_p.hpp>
+#include <stan/math/prim/scal/fun/inv.hpp>
+#include <stan/math/prim/scal/fun/digamma.hpp>
+#include <stan/math/prim/scal/fun/is_nan.hpp>
+#include <stan/math/prim/scal/fun/grad_reg_inc_gamma.hpp>
+#include <stan/math/prim/scal/err/domain_error.hpp>
+#include <limits>
+
+namespace stan {
+  namespace math {
+
+    template <int IMPL = 0>
+    struct lower_reg_inc_gamma {
+      template <int ARG, typename T1, typename T2>
+      static
+      typename boost::math::tools::promote_args<T1, T2>::type
+      grad(T1 a, T2 z, double precision = 1e-10, int max_steps = 1e5) {
+        typedef typename boost::math::tools::promote_args<T1, T2>::type TP;
+        TP d = std::numeric_limits<TP>::quiet_NaN;
+        return d;
+      }
+    };
+
+
+    template <>
+    struct lower_reg_inc_gamma<1> {
+      template <int ARG, typename T1, typename T2>
+      static
+      typename boost::math::tools::promote_args<T1, T2>::type
+      grad(T1 a, T2 z, double precision = 1e-10,
+                                 int max_steps = 1e5) {
+        typedef typename boost::math::tools::promote_args<T1, T2>::type TP;
+
+        if (is_nan(a)) return std::numeric_limits<TP>::quiet_NaN();
+        if (is_nan(z)) return std::numeric_limits<TP>::quiet_NaN();
```

Comment:
It's just what the current code does.  If I mess with it I have to adjust some auto-generated tests and I don't understand the purpose of the auto-generated tests.  If we had a clear policy on throwing versus not throwing I would happily fix it here and for other functions as well. Without a clear policy I don't see the point of trying to fix it and dealing with the tests.  Maybe it would be clearer if we talk at dev meeting.